kusto/internal/frames/unmarshal/json/README.md

```diff
@@ -0,0 +1,46 @@
+# Why the fork of encoding/json (from version 1.14)
+
+The json package is setup for the general use case. And it makes a lot of assumptions about how people are going to send you data.
+
+In our case, we want to read a stream of Kusto frames represented by a JSON array, but we want to read them as a stream and not all at once.
+
+That means we need to use a json.Decoder. Which is all well and dandy, except what we are going to read out varies by the type of frame. And we won't know what that frame is until we parse the frametype key.
+
+So we decode into a json.RawMessage, extract the frametype from the message, and then unmarshal into the concrete frame.  All good right?
+
+Wrong, because json.RawMessage makes a copy of the bytes that were sent in.  That costs us a huge amount in allocation per frame on 100's of thousands or millions of frames.
+
+Next, for some reason which I can't explain, Go translates numbers in a json.Unmarshal() call into float64 if the target is an interface{}.  I'm sure this is a good reason for it, I just don't understand it.  I would think it would just be an int64 or float64.  Or, because they made it, a json.Number.  The decoder itself can translate to json.Number, if you enable it.  But not Unmarshal.
```

Comment:
```suggestion
Next, for some reason which I can't explain, Go translates numbers in a json.Unmarshal() call into float64 if the target is an interface{}.  I'm sure there is a good reason for it, I just don't understand it.  I would think it would just be an int64 or float64.  Or, because they made it, a json.Number.  The decoder itself can translate to json.Number, if you enable it.  But not Unmarshal.
```